src/test/java/com/neuvector/ScannerTest.java

```diff
@@ -45,14 +64,26 @@ public void scanRegistryTest(){
         String repositoryTag = "3.6";
         String nvScannerImage = "neuvector/scanner:latest";
         String nvRegistryURL = registryURL;
+        //Replace by your own credentials
         String nvRegistryUser = "xxx";
         String nvRegistryPassword = "xxx";
         String mountPath = "/temp";  //mountPath is an optional parameter. It will use "/var/neuvector" by default.
-
         Registry registry = new Registry(registryURL, regUser, regPassword,repository,repositoryTag);
         NVScanner scanner = new NVScanner(nvScannerImage, nvRegistryURL, nvRegistryUser, nvRegistryPassword, mountPath);
 
         ScanRepoReportData scanReportData = Scanner.scanRegistry(registry, scanner, license);
-        assertTrue( scanReportData != null );
+        if(scanReportData.getError_message().isEmpty()) {
+            File file = new File(mountPath + "/scan_result.json");
+            Path path = Paths.get(mountPath + "/scan_result.json");
+            FileOwnerAttributeView fileOwner = Files.getFileAttributeView(path,
+                    FileOwnerAttributeView.class);
+            UserPrincipal user = fileOwner.getOwner();
+
+            assertTrue(!user.getName().equals("root"));
```

Comment:
If you revert your changes in `Scanner.java`, does this test fail at this point? In other words, will the username be `root` if you do not have your changes?

Reply:
I think so, but if the user is root `scanReportData.getError_message().isEmpty()`won't be empty because of the permissions error, and we will not get inside this block, but this test is just covering the current changes done

Reply:
If it is covering the current changes done, this test should fail without your changes. Otherwise there is no point in including this test IMO.

Reply:
Not sure about this point. The only thing I can test with this new changes is if the file was not created by root, was created by the current host user, do you have any suggestion? just to clarify better, thanks!!

Reply:
If you need to change the test itself to make the test fail when you remove your implementation, I think this is not an accurate test, isn't it? I mean if it is not failing without the implementation, what is it testing?

Reply:
Good point, IMO the current tests are reflecting that the code matches the reason of the new changes, I mean the code t is verifying, that the file was created even for root or by the host user, besides that the new behavior means that if the file does not exist, we must create a new one and it shouldn't be created by root and it is what the test is showing. Let me know your thoughts I am running out of ideas of how can I solve resolve the testing part, thanks!!